### PR TITLE
Changes in the best_calib selection criterion

### PR DESCRIPTION
**Pull request by emolinab**
## :bird: Description of this PR :bird:

- In this pull request the calc_calib.R was fixed to ensure that the reported calib_factors reported correspond to the divergence of the iteration, and to select the factors, per region, that correspond to the lowest divergence.

## :wrench: Checklist for PR creator :wrench:

- [x] Label pull request [from the label list](https://github.com/magpiemodel/magpie/labels).
  - **Low risk**: Simple bugfixes (missing files, updated documentation, typos) or changes  in start or output scripts
  - **Medium risk**: Uncritical changes in the model core (e.g. moderate modifications in non-default realizations)
  - **High risk**: Critical changes in model core or default settings (e.g. changing a model default or adjusting a core mechanic in the model)

- [x] Self-review own code
  - No hard coded numbers and cluster/country/region names.
  - The new code doesn't contain declared but unused parameters or variables.
  - [`magpie4`](https://github.com/pik-piam/magpie4) R library has been updated accordingly and backwards compatible where necessary.
  - `scenario_config.csv` has been updated accordingly (important if `default.cfg` has been updated)

- [x] Document changes 
  - Add changes to `CHANGELOG.md`
  - Where relevant, put In-code documentation comments
  - Properly address updates in interfaces in the module documentations
  - run [`goxygen::goxygen()`](https://github.com/pik-piam/goxygen) and verify the modified code is properly documented

- [x] Perform test runs
  - **Low risk**: 
    - Run a compilation check via `Rscript start.R --> "compilation check"`
  - **Medium risk**: 
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
  - **High risk**:
    - Run test runs via `Rscript start.R --> "test runs"`
    - Check logs for errors/warnings
    - Default run from the PR target branch for comparison
    - Provide relevant comparison plots (land-use, emissions, food prices, land-use intensity,...)

 **Comparison between old best_calib with the PR's best_calib (tag=Test2)**
1. Sticky- Improvement ( runs without land-conversion costs calibration)

image
   
2.  PerTon- Improvement ( runs without land-conversion costs calibration)
   
image

### :chart_with_downwards_trend: Performance changes :chart_with_upwards_trend:
  
  - Current develop branch default : ** mins
  - This PR's default :  ** mins

## :rotating_light: Checklist for reviewer :rotating_light:

- PR is labeled correctly
- Code changes look reasonable
  - No hard coded numbers and cluster/country/region names.
  - No unnecessary increase in module interfaces
  - model behavior/performance is satisfactory.
- Changes are properly documented
  - `CHANGELOG` is updated correctly
  - Updates in interfaces have been properly addressed in the module documentations
  - In-code documentation looks appropriate
- [ ] content review done (at least 1)
- [x] RSE review done (at least 1)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### fixed
 - **44_biodiversity** added regional layer `i` in `bii_target` realisation to make it compatible with the high-resolution parallel optimization output script
+-**scripts** fixed in the calc_calib.R script the saving of calib_factors used in each iteration to ensure that they correspond with the divergence reported. Changed divergence from zero to NA for those iterations where calib_factors are above the limit. The best_calib selection criterion was changed from selecting the factors of the iteration with the lowest standard deviation to the selection, for each region, of the factor of the iteration with the lowest divergence. Also, factors from the first iteration are now not considered, and if two different factors had the same divergence for a region, the one of the latest iteration is picked.
```
Comment (tscheypidi) [inline CHANGELOG.md:27]:
```suggestion
- **scripts** fixed in the calc_calib.R script the saving of calib_factors used in each iteration to ensure that they correspond with the divergence reported. Changed divergence from zero to NA for those iterations where calib_factors are above the limit. The best_calib selection criterion was changed from selecting the factors of the iteration with the lowest standard deviation to the selection, for each region, of the factor of the iteration with the lowest divergence. Also, factors from the first iteration are now not considered, and if two different factors had the same divergence for a region, the one of the latest iteration is picked.
```

**Review by tscheypidi — APPROVED** [review]
looks good overall, only thing I noticed is that the formatting of the script seems to be a bit off. I would recommend running `lucode2::lint("calc_calib.R")` and fix indentations (e.g. via CTRL+I in Rstudio)

**Review by tscheypidi — APPROVED** [review]
thanks for cleaning it

**Review by FelicitasBeier — APPROVED** [review]
Looks good to me.

**Review by pvjeetze — APPROVED** [review]
Look fine for me too.

**Pull request merged**